bot/cogs/Calculator.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import re
+import discord
+import discord.ext.commands as commands
+
+log = logging.getLogger(__name__)
+
+
+class Calculator(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self._last_member = None
+        self.operators = [
+            {'symbol': '+', 'precedence': 0, 'assoc': 'L'},
+            {'symbol': '-', 'precedence': 0, 'assoc': 'L'},
+            {'symbol': '*', 'precedence': 1, 'assoc': 'L'},
+            {'symbol': '/', 'precedence': 1, 'assoc': 'L'},
+            {'symbol': '^', 'precedence': 2, 'assoc': 'R'},
+        ]
+
+    @commands.command()
+    async def calc(self, ctx, *args):
+        await ctx.send('{} arguments: {}'.format(len(args), ', '.join(args)))
+
+        result = self.parse_postfix(self.parse_expression(args[0]))
+
+        await ctx.send(f'{args[0]} = {result}')
+
+    # compares the precedence of two operators
+    def compare_precedence(self, operator1, operator2):
+
+        op1 = self.search_operators_symbol(operator1)
+
+        op2 = self.search_operators_symbol(operator2)
+
+        return op1['precedence'] <= op2['precedence']
+
+    # searches through a list operators and return its information
+    def search_operators_symbol(self, symbol):
+        for operator in self.operators:
+            if symbol == operator['symbol']:
+                return operator
+
+    # checks if symbol is an operator
+    def is_operator(self, symbol):
+        for op in self.operators:
+            if symbol == op['symbol']:
+                return True
+        return False
+
+    # returns the last element in a list
+    def get_top_stack(self, stack):
+        return stack[len(stack) - 1]
+
+    def parse_expression(self, expression):
+
+        # parse expression into a list of numbers and symbols
+        tokens = re.findall("-?\d+|[+^/*()-]", expression)
+
+        output_que = []
+
+        operator_stack = []
+
+        for token in tokens:
+
+            intFlag = 1
+            try:
+                int(token)
+            except ValueError:
+                intFlag = 0
+
+            if intFlag:
+                output_que.append(token)
+
+            elif token == '(':
+                operator_stack.append(token)
+
+            elif token == ')':
+                # Ensures that proper precedence is followed with parentheses
+                while self.get_top_stack(operator_stack) != '(':
+                    output_que.append(self.get_top_stack(operator_stack))
+                    operator_stack.pop()
+                operator_stack.pop()
+
+            elif self.is_operator(token):
+                # Makes sure precedence is followed for operators
+                while (len(operator_stack) != 0 and (self.get_top_stack(operator_stack) not in "()") 
```

Comment:
I would break this if condition out into its own method with a succint name to clear up confusion about what this check does